docs/dev/reference/services.md

```diff
@@ -492,6 +492,51 @@ class Example
 }
 ```
 
+## InsertTagParser
+
+{{< version "4.13" >}}
+
+This service let you replace inserttags in your string content.
+
+
+```php
+use Contao\CoreBundle\InsertTag\ChunkedText;
+use Contao\CoreBundle\InsertTag\InsertTagParser;
+
+class Example
+{
+    private InsertTagParser $insertTagParser;
+
+    public function __invoke(string $buffer): string
+    {
+        // Return a string that may contain esi tags
```

Comment:
I think many people won't understand ESI. I don't think it's super helpful to describe what the methods do. It would be way more valuable to document **when** to use which method. So real world examples.

- `replace()` should be used in HTML context when the replaced result is sent as a response to the client. It will allow for ESI tags that can improve caching behaviour.
- `replaceInline()` should be used when the result is **not** sent to a client. Like e.g. when you use `$email->subject()` or so.

The chunked alternatives have very specific use cases which I think 99% of the developers will never need. But maybe @m-vo or @ausi can come up with a real world example here too?

Reply:
The example for chunked texts was taken from the twig ContaoExtension, I just replace twig_escape_filter with htmlspecialchars :)
I will adjust the texts, sound good :+1: